app/src/templates/quiz.html

```diff
@@ -0,0 +1,178 @@
+                        </label>
+                    </div>
+                    <p class="question">2. Test question</p>
+                    <div class="question-options">
+                        <label>
+                            <input
+                                type="radio"
+                                value="q2option1"
+                                name="test2"
+                                onclick="updateCount()"
+                            />
+                            Q2 Option 1
+                        </label>
+                        <label>
+                            <input
+                                type="radio"
+                                value="q2option2"
+                                name="test2"
+                                onclick="updateCount()"
+                            />
+                            Q2 Option 2
+                        </label>
+                        <label>
+                            <input
+                                type="radio"
+                                value="q2option3"
+                                name="test2"
+                                onclick="updateCount()"
+                            />
+                            Q2 Option 3
+                        </label>
+                        <label>
+                            <input
+                                type="radio"
+                                value="q2option4"
+                                name="test2"
+                                onclick="updateCount()"
+                            />
+                            Q2 Option 4
+                        </label>
+                    </div>
+                </div>
+
+                <button
+                    id="submit-quiz"
+                    type="submit"
+                    {%
+                    if
+                    user_has_submitted
+                    %}
+                    disabled
+                    {%
+                    endif
+                    %}
+                    class="clickable"
+                >
+                    Submit
+                </button>
+            </form>
+        </div>
+    </div>
+</main>
+{% endblock %} {% block scripts %}
+<script>
+    // Logic for enabling submit button
+    document.addEventListener("DOMContentLoaded", function () {
+        const submitButton = document.getElementById("submit-quiz");
+        const quizForm = document.getElementById("quiz-form");
+
+        // Initially disabled
+        submitButton.disabled = true;
+
+        // Add a change event listener to the form elements (assuming radio buttons for options)
+        quizForm.addEventListener("change", function () {
+            // Check if every button is pressed
+            const allQuestionsAnswered = Array.from(
+                document.querySelectorAll('.question input[type="radio"]')
+            ).every((input) => input.checked);
```

Comment:
Assuming you were trying to reference `.question-options`/`.question-block` in the line above and considering `input[type="radio"]`, I don't think `.every` would be the best option here. 

What this would do is check if _every single_ radio button was filled out (all 8), which is impossible to achieve given that only one option can be selected for every 4 radio buttons. As such, `allQuestionsAnswered` would always be false, and the button will never be enabled if checking for the value of this boolean below. 

One solution would be to:
- replace `document.querySelectorAll('.question input[type="radio"]')` with `document.querySelectorAll('.question-block input[type="radio"]:checked')`; this just filters for the number of radio buttons that are selected, instead of all radio inputs;
- while keeping the `Array.from`, use `.length` on it instead of `.every()` and test for whether if it is greater than or equal to your `totalCount` variable below. You're basically checking if there's an equal amount of filled in inputs for every question you have.